pkg/clusters/types/gke/builder.go

```diff
@@ -176,3 +194,67 @@ func (b *Builder) Build(ctx context.Context) (clusters.Cluster, error) {
 
 	return cluster, nil
 }
+
+// createCluster creates the GKE cluster asynchronously.
+func (b *Builder) createCluster(ctx context.Context, req *containerpb.CreateClusterRequest, mgrc *container.ClusterManagerClient, createdByID, authToken string) error {
+	// createSubnet is currently only available via gcloud CLI:
+	// https://github.com/googleapis/google-cloud-go/issues/7219
+	if b.createSubnet {
+		if err := b.createClusterUsingCLI(ctx, req, createdByID, authToken); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	_, err := mgrc.CreateCluster(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *Builder) createClusterUsingCLI(ctx context.Context, req *containerpb.CreateClusterRequest, createdByID, authToken string) error {
+	tokenFile, err := os.CreateTemp("", "gcloud-token-")
+	if err != nil {
+		return fmt.Errorf("failed to create a temporary file for gcloud token: %w", err)
+	}
+	defer func() {
+		_ = os.Remove(tokenFile.Name())
+	}()
+	if _, err := io.WriteString(tokenFile, authToken); err != nil {
+		return fmt.Errorf("failed to write a token to the temporary file: %w", err)
+	}
```

Comment:
I'm not saying that we need to do it but worth considering :) 

In bash (and at least in zsh too) you can use `<(command args)` to save the output of a command to a file descriptor and get its name.

So something like this is possible:

```bash
$ cat <(echo 123)
123
$ ls <(echo 123)
/dev/fd/11
```

In this case you could do something like

```
--access-token-file`, fmt.Sprtintf("<(echo %s)", authToken),
```

I haven't tried it with `exec.Command()` but worth a try.

Reply:
That's a nice idea, no luck unfortunately with making it happen. It worked neither with: 
- `exec.Command` directly
- running a shell using `exec.Command` (with `exec.Command("bash", "-c", rawCommand)`)
- running the command (`gcloud container clusters create t-f1e995d8-59ea-4517-97b4-b0e6a4b1b7b5 --access-token-file=<(echo "ya29.a0AX9GBd...)`) by hand in zsh (I should try that first 🙂) 

I understand that it's because gcloud CLI expects a real file and won't work with the process substitution as mentioned in this [SO answer](https://unix.stackexchange.com/a/64011). 😞